google/cloud/bigtable/examples/data_snippets.cc

```diff
@@ -944,6 +944,138 @@ void PopulateTableHierarchy(google::cloud::bigtable::Table table, int argc,
   }
 }
 
+void WriteSimple(google::cloud::bigtable::Table table, int argc, char*[]) {
+  if (argc != 1) {
+    throw Usage{"write-simple <project-id> <instance-id> <table-id>"};
+  }
+
+  // [START bigtable_writes_simple]
+  namespace cbt = google::cloud::bigtable;
+  [](cbt::Table table) {
+    auto timestamp = std::chrono::duration_cast<std::chrono::milliseconds>(
+        std::chrono::system_clock::now().time_since_epoch());
+
+    std::string row_key = "phone#4c410523#20190501";
+    cbt::SingleRowMutation mutation(row_key);
+    std::string column_family = "stats_summary";
+
+    mutation.emplace_back(cbt::SetCell(column_family, "connected_cell",
+                                       timestamp, std::int64_t{1}));
+    mutation.emplace_back(cbt::SetCell(column_family, "connected_wifi",
+                                       timestamp, std::int64_t{1}));
+    mutation.emplace_back(
+        cbt::SetCell(column_family, "os_build", timestamp, "PQ2A.190405.003"));
+    google::cloud::Status status = table.Apply(std::move(mutation));
+    if (!status.ok()) {
+      throw std::runtime_error(status.message());
+    }
+    std::cout << "Successfully wrote row" << row_key << "\n";
+  }
+  // [END bigtable_writes_simple]
+  (std::move(table));
+}
+void WriteBatch(google::cloud::bigtable::Table table, int argc, char*[]) {
+  if (argc != 1) {
+    throw Usage{"write-batch <project-id> <instance-id> <table-id>"};
+  }
+
+  // [START bigtable_writes_batch]
+  namespace cbt = google::cloud::bigtable;
+  [](cbt::Table table) {
+    auto timestamp = std::chrono::duration_cast<std::chrono::milliseconds>(
+        std::chrono::system_clock::now().time_since_epoch());
+    std::string column_family = "stats_summary";
+
+    cbt::BulkMutation bulk;
+    cbt::SingleRowMutation mutation("tablet#a0b81f74#20190501");
+    mutation.emplace_back(cbt::SetCell(column_family, "connected_cell",
+                                       timestamp, std::int64_t{1}));
+    mutation.emplace_back(
+        cbt::SetCell(column_family, "os_build", timestamp, "12155.0.0-rc1"));
+    bulk.emplace_back(std::move(mutation));
+    cbt::SingleRowMutation mutation2("tablet#a0b81f74#20190502");
+    mutation2.emplace_back(cbt::SetCell(column_family, "connected_cell",
+                                        timestamp, std::int64_t{1}));
+    mutation2.emplace_back(
+        cbt::SetCell(column_family, "os_build", timestamp, "12145.0.0-rc6"));
+    bulk.emplace_back(std::move(mutation2));
```

Comment:
Consider moving the definition of `bulk` here, where you could write:

```C++
cbt::BulkMutation bulk(std::move(mutation1), std::move(mutation2));
```

Alternatively, since `mutation` is not used after line 995 you could write something like:

```C++
cbt::BulkMutation bulk;

bulk.emplace_back(
  cbt::SingleRowMutation("tablet#....", cbt::SetCell(....), cbt::SetCell(...)));
```

All of them "work", but I think constructing the single row mutations in place is more idiomatic.